Sources/StreamChat/Database/DTOs/ChannelDTO_Tests.swift

```diff
@@ -124,6 +124,29 @@ class ChannelDTO_Tests: XCTestCase {
         }
     }
     
+    func test_DTO_hash_sameAsPayloadHash() throws {
+        let channelId: ChannelId = .unique
+        
+        let payload = ChannelDetailPayload.dummy(cid: channelId)
+        
+        // Save the payload to the db
+        try database.writeSynchronously { session in
+            try session.saveChannel(payload: payload, query: nil)
+        }
+        
+        // Load the channel from the db and check the fields are correct
+        var loadedChannel: ChannelDTO? {
+            database.viewContext.channel(cid: channelId)
+        }
+        
+        // Assert that hash is not changed
+        guard let dtoHash = loadedChannel?.chatHash else {
+            XCTFail("DTO is missing hash!")
+            return
+        }
+        XCTAssertEqual(Int(dtoHash), payload.chatHash)
+    }
+    
```

Comment:
Isn't there a missing test here? The one that actually checks the DTO is not saved when not needed? It seems to me that if I remove `if dto.chatHash != payload.chatHash {` from the DTO code everything will keep being green.

Reply:
Yes, since ChannelDetailPayload is a struct (Message and User payloads are classes) I couldn't subclass them to introduce `explicitHash`, I got stuck how to write a similar test tbh

Reply:
Suggestions welcome!